crates/proof_system/global_merkle_root/storage/src/structured_storage.rs

```diff
@@ -1,6 +1,18 @@
+use fuel_core_storage::Mappable;
+
 mod blobs;
 mod coins;
 mod contracts;
 mod messages;
 mod transactions;
 mod upgrades;
+
+/// Wrapper over an existing table to allow implementing new foreign traits on it.
+pub struct LocalMirror<Table>(core::marker::PhantomData<Table>);
+
+impl<Table: Mappable> Mappable for LocalMirror<Table> {
+    type Key = <Table as Mappable>::Key;
```

Comment:
```suggestion
    type Key = Table::Key;
```

curious if we need the explicit trait bound here if we've defined it above

Reply:
Good point, updated in https://github.com/FuelLabs/fuel-core/pull/2805/commits/bd9ee112de338754b40b0f0cd10d189da469cf24